src/OrleansRuntime/Core/IInvokable.cs

```diff
@@ -0,0 +1,12 @@
+using Orleans.CodeGeneration;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// Common internal interface for SystemTarget and ActivationData.
+    /// </summary>
+    internal interface IInvokable
+    {
+        IGrainMethodInvoker GetInvoker(GrainTypeManager typeManager, int interfaceId, string genericGrainType = null);
```

Comment:
@jdom I thought about that, but figured it would be too confusing to people reading the code. At the bottom of the stack, would you pass null? i.e, would `GrainTypeManager` ignore the `IInvokable` parameter?